azext_iot/iothub/providers/state.py

```diff
@@ -100,27 +117,76 @@ def process_hub(self, target: dict):
 
-            hub_state["devices"][id["deviceId"]] = identities[i]
+            identities[i]["properties"].pop("reported")
+            [identities[i]["properties"]["desired"].pop(key) for key in ["$metadata", "$version"]]
+            [identities[i].pop(key, None) for key in ["cloudToDeviceMessageCount", "configurations", "deviceEtag", "deviceScope",
+                                                      "lastActivityTime", "modelId", "parentScopes", "statusUpdateTime", "etag",
+                                                      "version"]]
+            hub_state["devices"]["identities"].append(identities[i])
 
             for module in module_objs:
                 module = vars(module)
 
                 if module["module_id"] not in ["$edgeAgent", "$edgeHub"]:
 
-                    # these can't be json serialized, and they can't be explicitly set anyways
-                    module.pop('connection_state_updated_time')
-                    module.pop('last_activity_time')
-
                     module_twin = _iot_device_module_twin_show(target=target, device_id=id["deviceId"],
                                                                module_id=module["module_id"])
 
                     module2 = _iot_device_module_show(target=target, device_id=id["deviceId"], module_id=module["module_id"])
                     module["authentication"] = module2["authentication"]
 
+                    [module.pop(key) for key in ["connection_state_updated_time", "last_activity_time", "etag",
+                                                 "cloud_to_device_message_count", "device_id"]]
+                    [module_twin.pop(key) for key in ["deviceEtag", "lastActivityTime", "etag", "version",
+                                                      "cloudToDeviceMessageCount", "statusUpdateTime"]]
+                    [module_twin["properties"]["desired"].pop(key) for key in ["$metadata", "$version"]]
+                    module_twin["properties"].pop("reported")
+
                     hub_state["modules"][id["deviceId"]].append([module, module_twin])
 
+        # CONTROL PLANE
+
+        if self.include_control_plane:
+            hub_name = target["entity"].split('.')[0]
+            rg = target["resourcegroup"]
+            control_plane = self.cli.invoke("iot hub show -n {} -g {}".format(hub_name, rg)).as_json()
+
+            hub_state["certificates"] = self.cli.invoke("iot hub certificate list --hub-name {} -g {}"
+                                                        .format(hub_name, rg)).as_json()["value"]
+
+            pbar = tqdm(total=len(hub_state["certificates"]), desc="Saving certificates", file=sys.stdout)
+            pbar.update(len(hub_state["certificates"]))
+            pbar.close()
+
+            hub_state["identities"] = control_plane["identity"]
+            num_identities = (1 if hub_state["identities"]["principalId"] else 0) + \
```

Comment:
its easier on the eyes if you have something like
```
num_identites = 0
if hub_state["identities"]["prinicipalId"]:
   num_identities += 1
if hub_state["identites"]["userAssignedIdentities"]:
  num_identites += len(hub_state["identites"]["userAssignedIdentities"])
```

if you want to shorten this, then I would shorten the first if statement but not both.